segments/gitstatus/gitstatus.p9k

```diff
@@ -0,0 +1,146 @@
+  # Register segment
+  # Parameters:
+  #   segment_name  context  background  foreground  Generic  Flat/Awesome-Patched  Awesome-FontConfig  Awesome-Mapped-FontConfig  NerdFont
+  p9k::register_segment  "GITSTATUS" "CLEAN"     "green"  "${DEFAULT_COLOR}" ''  ''  ''  ''  ''
+  p9k::register_segment  "GITSTATUS" "MODIFIED"  "yellow" "${DEFAULT_COLOR}" ''  ''  ''  ''  ''
+  p9k::register_segment  "GITSTATUS" "UNTRACKED" "green"  "${DEFAULT_COLOR}" '?' $'\uE16C' $'\uF059' '\u'${CODEPOINT_OF_AWESOME_QUESTION_CIRCLE} $'\uF059'
+
+  ################################################################
+  # Register segment default values
+  p9k::set_default P9K_GITSTATUS_MAX_INDEX_SIZE_DIRTY "4096"
+  p9k::set_default P9K_GITSTATUS_MAX_SYNC_LATENCY_SECONDS "5"
+  p9k::set_default P9K_GITSTATUS_SHOW_CHANGESET false
+  p9k::set_default P9K_GITSTATUS_CHANGESET_HASH_LENGTH 8
+  p9k::set_default P9K_GITSTATUS_ALWAYS_SHOW_REMOTE_BRANCH false
+  p9k::set_default P9K_GITSTATUS_HIDE_BRANCH_ICON false
+  p9k::set_default P9K_GITSTATUS_ACTIONFORMAT_FOREGROUND "001"
+}
+
+## Start the gitstatus daemon
+source ${P9K_GITSTATUS_DIR:-${_P9K_INSTALLATION_DIR}/gitstatus}/gitstatus.plugin.zsh
+gitstatus_start -m ${P9K_GITSTATUS_MAX_INDEX_SIZE_DIRTY} P9K
+
+################################################################
+# @description
+#   Displays the status of your git checkout.
+##
+# @args
+#   $1 string Alignment - left | right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+##
+prompt_gitstatus() {
+  gitstatus_query -d ${${GIT_DIR:a}:-$PWD} -t $P9K_GITSTATUS_MAX_SYNC_LATENCY_SECONDS P9K 2>/dev/null
+
+  # Print nothing, if we are not in a repo, or if the gitstatus daemon sends a timeout
+  [[ -z "${VCS_STATUS_RESULT}" || "${VCS_STATUS_RESULT}" =~ "norepo-" || "${VCS_STATUS_RESULT}" == "tmout" ]] && return
+
+  local current_state="CLEAN"
+  typeset -ah segment_content
+  if [[ "${P9K_GITSTATUS_SHOW_CHANGESET}" == "true" ]] && (( #VCS_STATUS_COMMIT )); then
+    segment_content+=("${__P9K_ICONS[GITSTATUS_COMMIT]}${VCS_STATUS_COMMIT[0,$P9K_GITSTATUS_CHANGESET_HASH_LENGTH]}")
+  fi
+  local localBranch="${VCS_STATUS_LOCAL_BRANCH}"
+  local remoteBranch="${VCS_STATUS_REMOTE_BRANCH}"
+  [[ -n "$localBranch" ]] && segment_content+=("${__P9K_ICONS[GITSTATUS_BRANCH]}${localBranch//\%/%%}")
+  # If no local branch and no remote branch was detected, we are probably in detached mode.
+  [[ -z "$localBranch" && -z "$remoteBranch" ]] && segment_content+=("${__P9K_ICONS[GITSTATUS_BRANCH]}(detached)")
+
+  if [[ -n "${remoteBranch}" \
+      && ( \
+        "${P9K_GITSTATUS_ALWAYS_SHOW_REMOTE_BRANCH}" == 'true' \
+        || "${remoteBranch#*/}" != "${VCS_STATUS_LOCAL_BRANCH#heads/}" \
+      ) ]]; then
+      segment_content+=("${__P9K_ICONS[GITSTATUS_REMOTE_BRANCH]}${remoteBranch//\%/%%}")
+  fi
+
+  (( VCS_STATUS_HAS_UNTRACKED )) && current_state="UNTRACKED" && segment_content+=("${__P9K_ICONS[GITSTATUS_UNTRACKED]}")
```

Comment:
VCS_STATUS_HAS_UNTRACKED and VCS_STATUS_HAS_UNSTAGED are tribools:

  0: no
  1: yes
  -1: unknown

-1 is returned if git index has more than P9K_GITSTATUS_MAX_INDEX_SIZE_DIRTY files

You probably want to handle 1 and -1 differently.

Reply:
👍